src/cats/SlideAnimation.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import clsx from "clsx";
+import styles from "./SlideAnimation.module.css";
+
+enum Direction {
+  Previous,
+  Next,
+}
+
+type SlideAnimationPropsType = {
+  child: JSX.Element;
+  direction: Direction;
+};
+
+function SlideAnimation({ child, direction }: SlideAnimationPropsType) {
+  const [slideChildren, setSlideChildren] = useState<Array<JSX.Element>>([]);
+
+  useEffect(() => {
+    setSlideChildren((prevChildren: Array<JSX.Element>) => {
+      const prevChild = prevChildren.slice(
+        ...(direction === Direction.Next ? [-1] : [0, 1])
+      );
+
+      return prevChild[0]?.key === child.key
+        ? prevChildren
+        : direction === Direction.Next
+        ? prevChild.concat([child])
```

Comment:
if prevChild is not an array, this can just become `[prevChild. child]`

Reply:
I can do that but need to change a little more logic to make that work. The current code doesn't make it obvious that the when prevChildren is empty it ends up returning [child]. 

It would look something like this:

`const mostRecent = prevChildren.at(direction === "next" ? -1 : 0);

  return mostRecent === undefined
    ? [child]
    : mostRecent.key === child.key
    ? prevChildren
    : direction === "next"
    ? [mostRecent, child]
    : [child, mostRecent];`